### ISSUE TITLE
Dart doc shows no result in auto complete for type names and only for static declarations inside them

### ISSUE THREAD

**Issue by FMorschel** [dart-lang/sdk#59724]
Repro:

lib.dart

```dart
class Class1 {
  static void foo() {}
  void bar() {}
}
```

main.dart

```dart
/// This is a dart doc.
/// It should show an auto-complete for Class1 over here: [C] but it doesn't.
/// Also should show autocomplete for instance values like bar over [Class1.] but it doesn't.
library main;
```

---

See here no items from Dart:

![image](https://github.com/user-attachments/assets/9cf237f7-9185-4857-a889-dbf2bbfdad36)

See here only triggering for static methods:

![image](https://github.com/user-attachments/assets/cdf31ec9-564b-4bb6-b1f2-616cec5f3012)

---

//CC @DanTup

**Comment by FMorschel** [comment]
Also, if this needs to add an import, we should agree with adding a doc import here too, I think (@bwilkerson).

**Comment by bwilkerson** [comment]
Yes. Doc imports are there for any names referenced only in documentation comments.

**Comment by FMorschel** [comment]
@srawlins comment https://github.com/dart-lang/sdk/issues/59910#issuecomment-2596155118:

> Dartdoc prefers the `A.foo` instance member. To specifically refer to the constructor, you can use `A.foo()`. We should document this behavior here: https://dart.dev/tools/doc-comments/references

So for this code, you can differentiate between them:

```dart
/// A class with [A.foo] and [A.foo()]
class A {
  int foo = 0;
  A.foo();
}
```

But there is no colouring for `A.foo()` (currently being suggested correctly). Should I open a new issue for this? Not sure where is the problem (@DanTup? I saw you fixed braces colouring previously somewhere).

![Image](https://github.com/user-attachments/assets/133057fb-ea1d-4131-854e-c6914582cbb4)

**Comment by srawlins** [comment]
Ah yeah, that'd be an Analysis Server issue I think.

**Comment by bwilkerson** [comment]
The test looks reasonable to me, at least on the surface.

I verified that the comment is associated with the class declaration, so it seems like it ought to be working.

The only advice I have is to use the advice in https://github.com/dart-lang/sdk/blob/main/pkg/analysis_server/doc/implementation/code_completion.md to debug the problem.

**Comment by FMorschel** [comment]
The completion suggestion gets all the way here but this regex is expecting different names on 207:

https://github.com/dart-lang/sdk/blob/5e8bd0243347ebd01c22d127e32dc58956bc97d5/pkg/analysis_server/test/client/completion_driver_test.dart#L197-L208

This is probably to avoid huge lists of suggestions here, I'll add this info to the docs for the tests here too.

**Comment by FMorschel** [comment]
Here is the initial CL to add the outline of tests for completion inside dart-docs https://dart-review.googlesource.com/410461.

@stereotype441 was suggested as reviewer by the CLI, so if you disagree, please assign someone else. I've added @srawlins because I've seen him working with Dart Docs before and being very familiar with the specs for it.

**Comment by FMorschel** [comment]
As a separate note, the above comment referencing auto-complete not showing up (quoted below), happens above:

- `library` (as noted above)
- `import` (maybe export and part as well but I'm unsure on why these would be commented, if you think I should add tests too, please say so)
- `extension`
- `typedef`

> For the above:
> 
> > /// It should show an auto-complete for Class1 over here: [C] but it doesn't.
> 
> similar to [#59946](https://github.com/dart-lang/sdk/issues/59946), this only happens at `library;`. It works fine over a class declaration.
> 
> ![Image](https://github.com/user-attachments/assets/36b2c336-e6d7-4145-a938-88779c93ba93)
> 
> ![Image](https://github.com/user-attachments/assets/81ad9f40-3fe9-4e22-8708-988eb3c44e14)

**Comment by FMorschel** [comment]
I already have a fix for `library` and `import` (and probably any directive as I wrote) here https://github.com/fmorschel/sdk/tree/auto-complete-base-tests.

**Comment by FMorschel** [comment]
Now last but not least, the fix for the instance members is at https://github.com/fmorschel/sdk/tree/auto-complete-instance. Once the base tests are merged I'll update these last messages with their respective CL links as soon as I open them. Thanks to everyone in this issue and the CL reviewers!

**Comment by FMorschel** [comment]
Alright, probably last update here. The three new CLs are open and updated above. The last one is medium sized and the other two are small ones. Thanks again everyone!
